modules/nf-lineage/src/main/nextflow/lineage/LinStoreFactory.groovy

```diff
@@ -51,15 +51,14 @@ abstract class LinStoreFactory implements ExtensionPoint {
         return factory.newInstance(config)
     }
 
-    static LinStore getOrCreate(Session session) {
+    static LinStore getOrCreate(LineageConfig config, boolean requireConfig=true) {
         if( instance || initialized )
             return instance
         synchronized (LinStoreFactory.class) {
             if( instance || initialized )
                 return instance
             initialized = true
-            final config = LineageConfig.create(session)
-            if( !config.enabled )
+            if( requireConfig && !config.enabled )
```

Comment:
I think this statement is not correct, if lineage is not enabled, it should return null for whatever `requireConfig` value.  When running the commands, requireConfig is false, and it always creates the config.